### PR TITLE
[Build] Provide correct rpath for macOS test bundles

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
This will allow test bundles to find any linked dylibs in the build
directory.

<rdar://problem/56793593>

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

File: Sources/Build/BuildPlan.swift [hunk 1]
```diff
@@ -1123,7 +1123,8 @@ public final class ProductBuildDescription {
         if buildParameters.triple.isLinux() {
             args += ["-Xlinker", "-rpath=$ORIGIN"]
         } else if buildParameters.triple.isDarwin() {
-            args += ["-Xlinker", "-rpath", "-Xlinker", "@loader_path"]
+            let rpath = product.type == .test ? "@loader_path/../../../" : "@loader_path"
```
Comment (abertelrud) [inline Sources/Build/BuildPlan.swift]: So this goes up out of the "MacOS" directory, then out of "Contents", then out of the test bundle, right?  i.e. it refers to the directory that contains the test bundle.

Comment (abertelrud) [reply Sources/Build/BuildPlan.swift]: If correct, then this looks good.

Comment (aciidb0mb3r) [reply Sources/Build/BuildPlan.swift]: Yep

**Pull request merged**